pkg/monitor/monitor/util.go

```diff
@@ -765,150 +778,158 @@ func getMonitorService(monitor *v1alpha1.TidbMonitor) []*core.Service {
 	reloaderPortName := "tcp-reloader"
 	prometheusPortName := "http-prometheus"
 	grafanaPortName := "http-grafana"
+	shards := monitor.GetShards()
+	for shard := int32(0); shard < shards; shard++ {
+		// currently monitor label haven't managedBy label due to 1.0 historical problem.
+		// In order to be compatible with 1.0 release monitor, we have removed managedBy label for now.
+		// We would add managedBy label key during released 1.2 version
+		var instanceName string
+		if shard == 0 {
+			instanceName = monitor.Name
+		} else {
+			instanceName = fmt.Sprintf("%s-shard-%d", monitor.Name, shard)
```

Comment:
What's the format of the different names? Some with `monitor` inserted, some are not.